openpype/hosts/maya/plugins/publish/validate_resolution.py

```diff
@@ -0,0 +1,81 @@
+import pyblish.api
+from openpype.pipeline import (
+    PublishValidationError,
+    OptionalPyblishPluginMixin
+)
+from maya import cmds
+from openpype.pipeline.publish import RepairAction
+from openpype.hosts.maya.api import lib
+from openpype.hosts.maya.api.lib import reset_scene_resolution
+
+
+class ValidateSceneResolution(pyblish.api.InstancePlugin,
+                              OptionalPyblishPluginMixin):
+    """Validate the scene resolution setting aligned with DB"""
+
+    order = pyblish.api.ValidatorOrder - 0.01
+    families = ["renderlayer"]
+    hosts = ["maya"]
+    label = "Validate Resolution"
+    actions = [RepairAction]
+    optional = True
+
+    def process(self, instance):
+        if not self.is_active(instance.data):
+            return
+        width, height, pixelAspect = self.get_db_resolution(instance)
+        current_renderer = cmds.getAttr(
+            "defaultRenderGlobals.currentRenderer")
+        layer = instance.data["renderlayer"]
+        if current_renderer == "vray":
+            vray_node = "vraySettings"
+            if cmds.objExists(vray_node):
+                control_node = vray_node
+                current_width = lib.get_attr_in_layer(
+                    "{}.width".format(control_node), layer=layer)
+                current_height = lib.get_attr_in_layer(
+                    "{}.height".format(control_node), layer=layer)
+                current_pixelAspect = lib.get_attr_in_layer(
+                    "{}.pixelAspect".format(control_node), layer=layer
+                )
+            else:
+                raise PublishValidationError(
+                    "Can't set VRay resolution because there is no node "
+                    "named: `%s`" % vray_node)
```

Comment:
```suggestion
                raise PublishValidationError(
                    "Can't detect VRay resolution because there is no node "
                    "named: `{}`".format(vray_node))
```